lib/posix-socket/socket.c

```diff
@@ -163,10 +165,33 @@ socket_ctl(const struct uk_file *sock, int fam, int req,
 	}
 }
 
+#define SOCKET_STATX_MASK \
+	(UK_STATX_TYPE|UK_STATX_MODE|UK_STATX_NLINK|UK_STATX_INO|UK_STATX_SIZE)
+
+static int
+socket_getstat(const struct uk_file *sock,
+	       unsigned int mask __unused, struct uk_statx *arg)
+{
+	/* Since all information is immediately available, ignore mask arg */
+	arg->stx_mask = SOCKET_STATX_MASK;
```

Comment:
What do you think of initializing `arg` with a compiler placed struct copy instead?
```c
arg = { .stx_mask = SOCKET_STATX_MASK,
        .stx_mode = S_IFSOCK|0777, 
        .stx_nlink = 1, 
        /* .... */
      };
```
My argument would be that this would make the code future proof, any later added field to `struct uk_statx` would be automatically zero'ed out.

Reply:
I had considered using designated initializers, however, since `struct statx` is pretty big (256 bytes at the time of writing) I'd rather explicitly _not_ have code fill in the entire thing if not needed, something explicitly allowed by the documentation (and is precisely what the mask arg/field is for).
Then again, Linux seems to zero out the output buffer regardless, an undocumented behavior apps may rely on, much like `stat`; plus stat(x) is usually not on the fastpath, so it'd make sense to be conservative here and zero out everything.

Question is: how do we ensure this for any and all stat calls, to maintain consistency? Designated initializers in the driver ensure zeroing out _only_ for that driver, but adding explicit `memset(0)` to `posix-fdio` may mean redundant zeroing out. We either
- (1) have `getstat` to use designated initializers in all drivers, enforced via code review (+ optional reminder comment in code) -- depends on human action, somewhat brittle, but should be fine once implemented and reviewers remember what's what
- (2) enforce zeroing out in `posix-fdio` -> drivers explicitly should _not_ use designated initializers to prevent doing double work -- more robust, worst case you lose performance but maintain compat

One other small caveat re: designated initializers: if we ever want this code to pass through a C++ compiler (super unlikely, since it's not a header and we're not crazy) we need to make sure the field order is _exactly_ the same as in the struct declaration (which atm it is not). C++ is dumb in that way.

Reply:
Hey, sorry, you're right, I somehow missed the mask and I agree with your chosen approach. Thinking about it again, I also agree that - as kernel - we should zero out the stat buffer for the `SYS_fstat` case (as you also patched in this PR), because with `fstat` you do not have a way to communicate which fields as application you are interested in and the kernel cannot indicate which components are supported by the subsystem. `statx` is of course a different consideration and it looks that we are actually close providing that system call, too. But overall, everything is fine for this PR. 👍🏻